bootstrap/src/main/java/io/airlift/bootstrap/ConfigurationException.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.airlift.bootstrap;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.inject.internal.Messages;
+import com.google.inject.spi.Message;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+
+public final class ConfigurationException
```

Comment:
This name collides with `com.google.inject.ConfigurationException`. Should we name it `BootstrapConfigurationException` or `ApplicationConfigurationException`? Or maybe it doesn't matter.

Reply:
I went with `ApplicationConfigurationException`